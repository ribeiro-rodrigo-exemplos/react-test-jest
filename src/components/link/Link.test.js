import React from 'react'
import Link from './Link'
import { render, fireEvent } from '@testing-library/react'

it('Link changes the class when hovered', () => {

    const { container, getByText } = render(<Link page="http://www.facebook.com">Facebook</Link>)

    //console.log(component)

    expect(getByText("Facebook")).toBeInTheDocument()
    //qexpect(container.firstChild).toMatchInlineSnapshot('')

    expect(container.firstChild.classList.contains('normal')).toBe(true)

    fireEvent.mouseEnter(container.firstChild)

    expect(container.firstChild.classList.contains('hovered')).toBe(true)

    fireEvent.mouseLeave(container.firstChild)

    expect(container.firstChild.classList.contains('normal')).toBe(true)

    expect(container).toMatchSnapshot()

})