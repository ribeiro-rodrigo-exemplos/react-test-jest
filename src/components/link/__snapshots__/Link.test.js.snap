// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div>
  <a
    class="normal"
    href="http://www.facebook.com"
  >
    Facebook
  </a>
</div>
`;
